### ISSUE TITLE
Stripe with React-Native 0.71.11

### ISSUE THREAD

**Issue by VambixDev** [stripe/stripe-react-native#1430]
**Describe the bug**
Got this error running .\gradlew build > 

> Task :stripe_stripe-react-native:lintDebug FAILEDs
> Lint found 28 errors, 197 warnings. First failure:
>
> C:\temp\Project\node_modules\@stripe\stripe-react-> native\android\src\main\java\com\reactnativestripesdk\CardFieldView.kt:215: Error: Companion.create can only be called from > within the same library group (referenced groupId=com.stripe from groupId=Vambix) [RestrictedApi]
>     val countryCode = CountryCode.create(value = countryString ?: LocaleListCompat.getAdjustedDefault()[0]?.country ?: "US")
>                                    ~~~~~~
>
> The full lint text report is located at:
>  C:\temp\Project\node_modules\@stripe\stripe-react-> native\android\build\intermediates\lint_intermediate_text_report\debug\lint-results-debug.txt
>: Detected multiple Kotlin daemon sessions at build\kotlin\sessions
>
>FAILURE: Build failed with an exception.
>
>* What went wrong:
>Execution failed for task ':stripe_stripe-react-native:lintDebug'.
>                 Lint found errors in the project; aborting build.
>
>                  Fix the issues identified by lint, or create a baseline to see only new errors:
>                  ```
>                  android {
>                      lint {
>                          baseline = file("lint-baseline.xml")
>                      }
>                  }
>                  ```
>
>                  For more details, see https://developer.android.com/studio/write/lint#snapshot
>
>                  Lint found 28 errors, 197 warnings. First failure:
>
>  C:\temp\Project\node_modules\@stripe\stripe-react->native\android\src\main\java\com\reactnativestripesdk\CardFieldView.kt:215: Error: Companion.create can only be called from >within the same library group (referenced groupId=com.stripe from groupId=Vambix) [RestrictedApi]
>        val countryCode = CountryCode.create(value = countryString ?: LocaleListCompat.getAdjustedDefault()[0]?.country ?: >"US")
>                                      ~~~~~~
>
>  The full lint text report is located at:
>    C:\temp\Project\node_modules\@stripe\stripe-react->native\android\build\intermediates\lint_intermediate_text_report\debug\lint-results-debug.txt
>
>* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
>
>* Get more help at https://help.gradle.org
>
>BUILD FAILED in 7m 25s
>792 actionable tasks: 672 executed, 120 up-to-date
>
>
**To Reproduce**
1. Run ./gradlew build in android folder

**Expected behavior**
Build success

**Screenshots**
![image](https://github.com/stripe/stripe-react-native/assets/134436251/f6a04afd-d3e0-4374-ba23-a4339f798975)

**Additional context**
This project is just created from termplate (npx react-native init) and cannot build after adding stripe-react-native package.

My package.json:

![image](https://github.com/stripe/stripe-react-native/assets/134436251/2f419064-07d0-456c-ba62-66a3dd2b96ad)

Parameters of build.gradle:
        buildToolsVersion = "33.0.0"
        minSdkVersion = 21
        compileSdkVersion = 33
        targetSdkVersion = 33

        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
        ndkVersion = "23.1.7779620"

        classpath("com.android.tools.build:gradle:7.4.2")
         
        gradle-wrapper.properties:
        distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-all.zip

Reading [this](https://developer.android.com/reference/tools/gradle-api/7.4/com/android/build/api/dsl/Lint) documentation 
I added this section of code in my android/app/build.gradle file:

>lint {
>         quiet = true
>         abortOnError = false
>         disable += ['RestrictedApi','NewApi'] //tried this to fix
>         checkReleaseBuilds = false
>         ignoreWarnings = true
>         baseline file("lint-baseline.xml")
>         textReport = true
>         ignore 'TypographyQuotes'
>         ignoreTestSources true
>         checkGeneratedSources true
>         checkDependencies false
>    }

**Comment by VambixDev** [comment]
I am not able disable linting issues but it not prevent the project from actually building project. Due to it i close the issue.

**Comment by geekdroid07** [comment]
i have the same issue

**Comment by hubertjaruzal** [comment]
Same problem, @charliecruzan-stripe adding lint options doesn't solve that.

```
    lint {
        quiet true
        checkReleaseBuilds false
        //If you want to continue even if errors found use following line
        abortOnError false
    }
```

**Comment by CrashingTheCode** [comment]
I have the exact same issue and I look into the logs and I see this: 
`/node_modules/@stripe/stripe-react-native/android/src/main/java/com/reactnativestripesdk/utils/Mappers.kt:810: Error: Call requires API level 24 (current min is 21): java.lang.Iterable#forEach [NewApi]
  setupIntent.paymentMethodTypes.forEach { code ->
                                 ~~~~~~~

   Explanation for issues of type "NewApi":
   This check scans through all the Android API calls in the application and
   warns about any calls that are not available on all versions targeted by
   this application (according to its minimum SDK attribute in the manifest).

   If you really want to use this API and don't need to support older devices
   just set the minSdkVersion in your build.gradle or AndroidManifest.xml
   files.

   If your code is deliberately accessing newer APIs, and you have ensured
   (e.g. with conditional execution) that this code will only ever be called
   on a supported platform, then you can annotate your class or method with
   the @TargetApi annotation specifying the local minimum SDK to apply, such
   as @TargetApi(11), such that this check considers 11 rather than your
   manifest file's minimum SDK as the required API level.

   If you are deliberately setting android: attributes in style definitions,
   make sure you place this in a values-vNN folder in order to avoid running
   into runtime conflicts on certain devices where manufacturers have added
   custom attributes whose ids conflict with the new ones on later platforms.

   Similarly, you can use tools:targetApi="11" in an XML file to indicate that
   the element will only be inflated in an adequate context.`

**Comment by malinjr07** [comment]
Stuck with it, too.

**Comment by charliecruzan-stripe** [comment]
Could you try adding `@SuppressLint("RestrictedApi")` right above this function declaration- https://github.com/stripe/stripe-react-native/blob/master/android/src/main/java/com/reactnativestripesdk/CardFieldView.kt#L220

and rebuild?

**Comment by malinjr07** [comment]
> Could you try adding `@SuppressLint("RestrictedApi")` right above this function declaration- https://github.com/stripe/stripe-react-native/blob/master/android/src/main/java/com/reactnativestripesdk/CardFieldView.kt#L220
> 
> and rebuild?

This code might be in node_modules, isn't it? Will changing that code can be a solution at all? Because each time I'll install any package or setup the app, I'll have to do that repeatedly.

**Comment by malinjr07** [comment]
This issue is happening with the latest version of React Native as well. I wish @charliecruzan-stripe could notice this and give an immediate solution.

**Comment by charliecruzan-stripe** [comment]
>This code might be in node_modules, isn't it?

Correct, but it would unblock you until we release an official fix. This is also just a lint warning so it shouldn't prevent you from actually building your app

**Comment by malinjr07** [comment]
Alright, I might not be an Official Engineer of Stripe (I wish they would hire me!). But Here's the Direct and Stable Fix until they patch the bug fix and launch the new version:

Instead of `./gradlew clean build` or `./gradlew clean` or `./gradlew build` use the following:

`./gradlew clean build` => `./gradlew clean build  -x lint`
`./gradlew clean` => `./gradlew clean  -x lint`
`./gradlew build` => `./gradlew build  -x lint`

This command will help you with both **RestrcitedAPI** and **NewAPI** error.

**Comment by rakshitbharat** [comment]
its now 2024 and
    "react-native": "^0.73.2",
    
    still this issue is in this world

**Comment by charliecruzan-stripe** [comment]
Could you try adding `@SuppressLint("RestrictedApi")` right above this function declaration- https://github.com/stripe/stripe-react-native/blob/master/android/src/main/java/com/reactnativestripesdk/CardFieldView.kt#L220

and rebuild?

that should be a patch to fix it and I'll open a PR with the fix

**Comment by Hiba-Bazerbashi** [comment]
I have the same error with react-native-gesture-handler , 

"react-native": "0.77.1",
"react-native-gesture-handler": "^2.24.0",

is there any fix ????

**Comment by jaynewstrom-stripe** [comment]
Hi @Hiba-Bazerbashi please open a new issue with a description of your issue, and steps to reproduce.
